shimmingtoolbox/load_nifti.py

```diff
@@ -0,0 +1,168 @@
+    else:
+        select_run = list(run_list.keys())[0]
+        logging.info(f"Reading acquisitions for run {list(run_list.keys())[0]}")
+
+    # Create output array and headers
+    nifti_pos = 0
+    echo_shape = sum(1 for tmp_info in run_list[select_run] if modality in tmp_info[1])
+    if nifti_init[0][0].ndim == 3:
+        niftis = np.empty([nifti_init[0][0].shape[0], nifti_init[0][0].shape[1], nifti_init[0][0].shape[2], echo_shape, 1], dtype=float)
+        for i_echo in range(len(run_list[select_run])):
+            tmp_nii = run_list[select_run][i_echo][0]
+            if modality in run_list[select_run][i_echo][1]:
+                info.append(tmp_nii[0].header)
+                json_info.append(tmp_nii[1])
+                niftis[:, :, :, nifti_pos, 0] = tmp_nii[2]
+                nifti_pos += 1
+    else:
+        niftis = np.empty([
+            nifti_init[0][0].shape[0], nifti_init[0][0].shape[1], nifti_init[0][0].shape[2],
+            echo_shape, nifti_init[0][0].shape[3]], dtype=float)
+        for i_echo in range(len(run_list[select_run])):
+            if modality in run_list[select_run][i_echo][1]:
+                tmp_nii = run_list[select_run][i_echo][0]
+                info.append(tmp_nii[0].header)
+                json_info.append(tmp_nii[1])
+                for i_volume in range(nifti_init[0][0].shape[3]):
+                    niftis[:, :, :, nifti_pos, i_volume] = tmp_nii[2][:, :, :, i_volume]
+                nifti_pos += 1
+
+    return niftis, info, json_info
+
+
+def read_nii(nii_path, auto_scale=True):
+    """ Reads a nifti file and returns the corresponding image and info. Also returns the associated json data.
+    Args:
+        nii_path (str): direct path to the .nii or .nii.gz file that is going to be read
+        auto_scale (:obj:`bool`, optional): Tells if scaling is done before return
+    Returns:
+        info (Nifti1Image): Objet containing various data about the nifti file (returned by nibabel.load)
+        json_data (dict): Contains the different fields present in the json file corresponding to the nifti file
+        image (ndarray): Image contained in the read nifti file. Siemens phase images are rescaled between 0 and 2pi.
+    """
+
+    info = nib.load(nii_path)
+
+    # `extractBefore` should get the correct filename in both.nii and.nii.gz cases
+    json_path = nii_path.split('.nii')[0] + '.json'
+
+    if os.path.isfile(json_path):
+        json_data = json.load(open(json_path))
+    else:
+        raise ValueError('Missing json file')
+
+    image = np.asarray(info.dataobj)
+    if auto_scale:
+        if ('Manufacturer' in json_data) and (json_data['Manufacturer'] == 'Siemens') \
+                and ("phase" in json_data['ImageComments']):
+            image = image * (2 * math.pi / PHASE_SCALING_SIEMENS)
+
+    return info, json_data, image
+
+
+if __name__ == "__main__":
+    load_nifti("C:\\Users\\Gabriel\\Documents\\share\\008_a_gre_DYNshim1")
```

Comment:
This will obviously need to be removed before merging

Reply:
Yes it was while I was testing